lib/haxe/src/org/apache/thrift/transport/THttpClient.hx

```diff
@@ -94,10 +114,241 @@ class THttpClient extends TTransport {
+                me.onError("Unknown host");
+            default:
+                me.req = null;
+                me.responseData = r.responseText;
+                me.onError("Http Error #"+r.status);
+            }
+        };
+        if( async )
+            r.onreadystatechange = onreadystatechange;
+        var uri = postData;
+        var jsData = binaryPostData;
+        if( jsData != null )
+            post = true;
+        else for( p in params ) {
+            if( uri == null )
+                uri = "";
+            else
+                uri += "&";
+            uri += StringTools.urlEncode(p.param)+"="+StringTools.urlEncode(p.value);
+        }
+        try {
+            if( post )
+                r.open("POST",url,async);
+            else if( uri != null ) {
+                var question = url.split("?").length <= 1;
+                r.open("GET",url+(if( question ) "?" else "&")+uri,async);
+                uri = null;
+            } else
+                r.open("GET",url,async);
+        } catch( e : Dynamic ) {
+            me.req = null;
+            onError(e.toString());
+            return;
+        }
+
+        //XHR binary charset opt by Marcus Granado 2006 [http://mgran.blogspot.com]
+        req.overrideMimeType("text\\/plain; charset=x-user-defined");
+
+        #if (haxe_ver >= 3.3)
+        r.withCredentials = me.withCredentials;
+        #end
+        if( !Lambda.exists(headers, function(h) return h.header == "Content-Type") && post && postData == null )
+            r.setRequestHeader("Content-Type","application/x-www-form-urlencoded");
+
+        for( h in headers )
+            r.setRequestHeader(h.header,h.value);
+
+        if( jsData != null ) {
+            r.send(new js.html.Uint8Array(jsData.getData()));
+        } else {
+            r.send(uri);
+        }
+
+        if( !async )
+            onreadystatechange(null);
+    }
+    #elseif nodejs
+    public override function request( ?post : Bool ) : Void {
+
+        var url_regexp = ~/^(https?:\/\/)?([a-zA-Z\.0-9_-]+)(:[0-9]+)?(.*)$/;
+        if( !url_regexp.match(url) ) {
+            onError("Invalid URL");
+            return;
+        }
+        var ssl = (url_regexp.matched(1) == "https://");
+        var host = url_regexp.matched(2);
+        var portString = url_regexp.matched(3);
+        var request = url_regexp.matched(4);
+        if( request == "" )
+            request = "/";
+        var port = if ( portString == null || portString == "" ) ssl ? 443 : 80 else Std.parseInt(portString.substr(1, portString.length - 1));
+
+        var options = {
+            "hostname":host,
+            "port":port,
+            "path":request,
+            "method":post ? "POST" : "GET",
+            "agent":false,
+            "withCredentials":#if (haxe_ver >= 3.3) me.withCredentials #else false #end
+        };
+
+        var headersCode = "";
+        for( h in headers )
+            headersCode += 'req.setHeader("${h.header}","${h.value}");';
+
```

Comment:
access to request_ and headers is 'private' for THttpClient. so there no way to pass extra headers into code 

of course jsHttp type isn't safe , but it isn't part of public api